src/charm.py

```diff
@@ -72,57 +72,177 @@ def __init__(self, *args) -> None:
             self._certificates.on.certificate_expiring, self._on_certificate_expiring
         )
 
-    def _configure_ausf(
-        self,
-        event: EventBase,
-    ) -> None:
-        """Configure AUSF configuration file and pebble service.
+    def _on_collect_unit_status(self, event: CollectStatusEvent):
+        """Check the unit status and set to Unit when CollectStatusEvent is fired.
 
         Args:
-            event (EventBase): Juju event
+            event: CollectStatusEvent
         """
+        if not self.unit.is_leader():
+            # NOTE: In cases where leader status is lost before the charm is
+            # finished processing all teardown events, this prevents teardown
+            # event code from running. Luckily, for this charm, none of the
+            # teardown code is necessary to perform if we're removing the
+            # charm.
+            event.add_status(BlockedStatus("Scaling is not implemented for this charm"))
+            return
+
         if not self._container.can_connect():
-            self.unit.status = WaitingStatus("Waiting for container to start")
+            event.add_status(WaitingStatus("Waiting for container to start"))
             return
+
         for relation in ["fiveg_nrf", "certificates"]:
             if not self._relation_created(relation):
-                self.unit.status = BlockedStatus(f"Waiting for {relation} relation")
+                event.add_status(BlockedStatus(f"Waiting for {relation} relation"))
                 return
+
         if not self._nrf_data_is_available:
-            self.unit.status = WaitingStatus("Waiting for NRF data to be available")
+            event.add_status(WaitingStatus("Waiting for NRF data to be available"))
             return
+
         if not self._container.exists(path=CONFIG_DIR):
-            self.unit.status = WaitingStatus("Waiting for storage to be attached")
+            event.add_status(WaitingStatus("Waiting for storage to be attached"))
             return
         if not _get_pod_ip():
-            self.unit.status = WaitingStatus("Waiting for pod IP address to be available")
+            event.add_status(WaitingStatus("Waiting for pod IP address to be available"))
+            return
+
+        if self._csr_is_stored() and not self._get_current_provider_certificate():
+            event.add_status(WaitingStatus("Waiting for certificates to be stored"))
+            return
+
+        if not self._ausf_service_is_running():
+            event.add_status(WaitingStatus("Waiting for AUSF service to start"))
+            return
+
+        event.add_status(ActiveStatus())
+
+    def _ausf_service_is_running(self) -> bool:
+        """Check if the AUSF service is running."""
+        try:
+            self._container.get_service(service_name=self._service_name)
```

Comment:
Does this really check whether the service is running or only is it defined in a pebble plan?
I see that pebble has a method called `is_running()` for checking the service state.

Reply:
Called is_running() to check the service status.

Reply:
Done